scripts/build_podspecs.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+
+# Copyright 2020, gRPC Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+import random
+import string
+import argparse
+
+class Dependency:
+    def __init__(self, name, version="s.version.to_s", internal=True):
+        self.name = name
+        self.version = version
+        self.internal = internal
+
+    def as_podspec(self):
+        if self.internal:
+            return "    s.dependency '%s', %s\n" % (self.name, self.version)
+        else: 
+            return "    s.dependency '%s', '%s'\n" % (self.name, self.version)
+
+class Pod:
+    def __init__(self, name, module_name, version, dependencies=[]):
```

Comment:
This won't work the way you expect it to since `dependencies=[]` is only evaluated once; all instances of `Pod` which rely on the default parameter will share the `dependencies` instance (I don't think this is an issue here since you only rely on the default for CGRPCZlib but it's a bug waiting to happen!). You can use `None` as the default and check against that instead.

Here's a minimal repro:

```python
class Foo:
    def __init__(self, dependencies=[]):
        self.dependencies = dependencies

f = Foo()
g = Foo()
g.dependencies.append(1)

print(f.dependencies)  # prints "[1]"
```

Reply:
For reference: https://docs.python.org/3/reference/compound_stmts.html#function-definitions

Reply:
Oh wow, I actually didn't know that fun fact! (Maybe I did and just forgot it over the years) Great catch! 👏

Reply:
Now that I'm reading about it more.... that's very strange behavior!

Reply:
Yeah, the behaviour is a little surprising.